inst/chunks/fun_scale_time_err.stan

```diff
@@ -21,26 +21,32 @@
    }
   /* scale and correlate time-series residuals
    * allowx for flexible correlation matrix subsets
-   * Args:
-   *   zerr: standardized and independent residuals
-   *   sderr: standard deviation of the residuals
-   *   chol_cor: cholesky factor of correlation matrix
-   *   nobs: number of observations in each group
-   *   begin: the first observation in each group
-   *   end: the last observation in each group
+   * Deviating Args:
    *   Jtime: array of time indices per group
    * Returns:
    *   vector of scaled and correlated residuals
    */
    vector scale_time_err_flex(vector zerr, real sderr, matrix chol_cor,
                               int[] nobs, int[] begin, int[] end, int[,] Jtime) {
      vector[rows(zerr)] err;
-     matrix[rows(chol_cor), cols(chol_cor)] Cor;
-     Cor = multiply_lower_tri_self_transpose(chol_cor);
-     for (i in 1:size(nobs)) {
+     matrix[rows(chol_cor), cols(chol_cor)] Cor = multiply_lower_tri_self_transpose(chol_cor);
+     int I = size(nobs);
+     vector[I] lp;
+     int has_lp[I] = rep_array(0, I);
+     int i = 1;
+     while (sum(has_lp) != I) {
        int iobs[nobs[i]] = Jtime[i, 1:nobs[i]];
-       err[begin[i]:end[i]] =
-         sderr * cholesky_decompose(Cor[iobs, iobs]) * zerr[begin[i]:end[i]];
-     }
-     return err;
-   }
+       matrix[nobs[i], nobs[i]] L = cholesky_decompose(Cor[iobs, iobs]);
```

Comment:
uhh..hurts to read this... maybe we can have an if statement for identifying the case whenever iobs[i] == 1,2,3,...,rows(chol_cor)? In that case you can just take L=chol_cor and can avoid a full cholesky_decompose, which is super expensive and this is probably even a rather common case?